viper.go

```diff
@@ -40,13 +40,19 @@ import (
 
 var v *Viper
 
+type Response struct {
```

Comment:
I think we should rename this to `RemoteResponse`; which is in line with `RemoteProvider`, and it prevents people to confuse it with something they should care about. Fix that and I will merge.